### PR TITLE
style resources page

### PR DESCRIPTION
**Pull request by larrycoal**
## Description

This PR style the resource page and added gdsc logo to the header

## Issue

This PR fixes #\17

## Preview

https://gdsc-larry-demo.netlify.app/

### CODE REVIEW

File: src/constant.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import gdscLogo from "./assets/images/logo_GDSC.png";
```
Comment (ANDREYDEN) [inline src/constant.js]: Good use of a constant file! I would advocate putting this file into a `src/utils/` folder since we'll probably have more helper files like this to come.

File: src/components/menu/Menu.css [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
-  justify-content: center;
+  font-size: 2rem;
+}
+.menu-header img {
+  height: 50px;
+  width: 50px;
+}
+.menu-item {
+  margin: 0 1rem;
+  text-decoration: none;
```
Comment (ANDREYDEN) [inline src/components/menu/Menu.css]: Nice work with the styling! Keep in mind though that we want to modularize our work and not exceed the boundaries of the targeted issue. I'll allow it this time, but for future reference, if you want to modify something else (like styling the navigation menu), please open a new issue so that you/other contributers can work on it

File: src/constant.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    link: "https://conestogastudents.com/2021/09/01/coding-club/",
+    description: "You can register for the official GDSC Conestoga CSI club",
+    logo: csiLogo,
+  },
+  {
+    name: "YouTube",
+    link: "https://www.youtube.com/channel/UCBbpXl_fLG6dbu25nu2nRNA",
+    description: "You can watch all of our event on our youtube channel",
+    logo: youtubeLogo,
+  },
```
Comment (ANDREYDEN) [inline src/constant.js]: The [doc](https://docs.google.com/document/d/1fkaTzob9xHglfr85g30PdW8Nv7wcwz9Doi1lZC_0rbE/edit?usp=sharing) we are aiming to replace has 7 (6 without the calendar) links with useful resources. Let's include all of them here. Let me know if you need any help with icons/text content.

File: src/pages/resources/Resources.css [hunk 4]
```diff
@@ -1,3 +1,44 @@
...
+}
+.top {
+  text-align: center;
+  padding: 4rem;
+}
+.bottom {
+  width: 80%;
+  margin: 0 auto;
+}
+.resourcesList {
```
Comment (ANDREYDEN) [inline src/pages/resources/Resources.css]: I don't have a strong opinion about this since we are not following any style guide here (yet), but usually there's a blank line between CSS rule blocks. No need to change anything, just FYI

File: public/index.html [hunk 5]
```diff
@@ -10,6 +10,13 @@
       content="Web site created using create-react-app"
     />
     <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+    <link
+      rel="stylesheet"
+      href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0/css/all.min.css"
+      integrity="sha512-9usAa10IRO0HhonpyAIVpjrylPvoDwiPUiKdWk5t3PyolY1cOd4DSE0Ga+ri4AuTroPR5aQvXU9xC6qOPnzFeg=="
+      crossorigin="anonymous"
+      referrerpolicy="no-referrer"
+    />
```
Comment (ANDREYDEN) [inline public/index.html]: Why do we need this link? Did I miss you using Font Awesome somewhere?

File: src/pages/resources/Resources.js [hunk 6]
```diff
@@ -1,24 +1,38 @@
...
-          explanation='CSI Club - register for the official GDSC Conestoga CSI club' />
-        <ExternalLink 
-          url='https://www.youtube.com/channel/UCBbpXl_fLG6dbu25nu2nRNA' 
-          explanation='YouTube - our channel with event recordings' />
+    <div className="resources_wrapper">
+      <div className="top">
+        <h1>OUR RESOURCES</h1>
+      </div>
+      <div className="bottom">
+        <ul className="resourcesList"> {showResources()}</ul>
```
Comment (ANDREYDEN) [inline src/pages/resources/Resources.js]: IMHO creating a separate constant configuration list and then mapping over it within a function is less readable than what we had previously: a hardcoded list of components with inline values. I would see value in your approach if we would have a 100 resources coming from some external API - in this case mapping over a dynamic list makes most sense. I'm not married to this idea though, so you are welcome to prove me wrong in the comments if you feel strongly about your aproach.

**Review by ANDREYDEN — CHANGES_REQUESTED** [review]
Awesome work, your changes look very tidy! A thing to keep in mind (which might be annoying, but is easily solvable in your case) is that we need to make sure our website is responsive and looks good on mobile as well. For the resources page a good solution would be going for a multi-column layout for desktop and single-column layout for mobile.

File: public/index.html (hunk 5 above)
Comment (larrycoal) [reply public/index.html]: using a link icon in the resources page, and i believe will be useful as we continue to build

File: src/pages/resources/Resources.js (hunk 6 above)
Comment (larrycoal) [reply src/pages/resources/Resources.js]: I think its easier to edit our resources if its kept in a separate file, and just makes our html shorter

File: public/index.html (hunk 5 above)
Comment (ANDREYDEN) [reply public/index.html]: ok, I see, do you mind switching it for the [Material Icons](https://mui.com/components/icons/) since that better fits into our Google theme?

File: src/pages/resources/Resources.js (hunk 6 above)
Comment (ANDREYDEN) [reply src/pages/resources/Resources.js]: ok, let's keep the function syntax, but in that case I'll ask you to do 2 things:
1. Clean up the `ExternalLink` component since it's not being used anywhere now
2. Inline the `map` on line 32 and make the function only return one resource (`showResource`). I think this way would be the best of 2 worlds

**Review by ANDREYDEN — CHANGES_REQUESTED** [review]
A few more things:
- for some reason I get 2 scrollbars on desktop:
image
- please use png/svg for the GitHub and YouTube icons so we don't get the white background:
image
- something is off with the sizing (width) of the icons:
  - desktop: desktop
  - mobile: mobile

File: src/index.css [hunk 7]
```diff
@@ -1,26 +1,28 @@
...
+body,
+#root {
+  height: 100vh;
 }
-
+/* temporary till we add hamburger, its messing with page styling as it was */
 header {
-  height: 5vh;
+  overflow: scroll;
 }
```
Comment (ANDREYDEN) [inline src/index.css]: Can you elaborate on why is this needed? I'm getting this result (unnecessary scroll):
image

File: src/pages/resources/Resources.js [hunk 8]
```diff
@@ -1,24 +1,33 @@
-import ExternalLink from "../../components/externalLink/ExternalLink"
+import React from "react";
+import "./Resources.css";
+import { resources } from "../../utils/constant";
 
-function Resources() {
+const Resources = () => {
```
Comment (ANDREYDEN) [inline src/pages/resources/Resources.js]: Why did you decide to go with an arrow function syntax? All other components use `function`

File: src/pages/resources/Resources.js [hunk 9]
```diff
@@ -1,24 +1,33 @@
...
+              </div>
+              <div className="header">{resource.name}</div>
+              <div>{resource.description}</div>
+              <div className="link">
+                <a href={resource.link} target="_blank" rel="noreferrer">
+                  {resource.name}
+                </a>
+              </div>
+            </li>
+          ))}
```
Comment (ANDREYDEN) [inline src/pages/resources/Resources.js]: You can stil extract this into a function, if you want. What I meant was inlining the `map` but keeping the `renderResourceFunction()`
```suggestion
          {resources.map(renderResource)}
```

**Review by ANDREYDEN — CHANGES_REQUESTED** [review]
Almost there, there are still changes to be done with the icons:
- the YouTube icon is a bit weird, is this the official one?
image
- Icon sizes are inconsistent between screen sizes (let's make them always square):
image

File: src/index.css (hunk 7 above)
Comment (larrycoal) [reply src/index.css]: Its not needed, i just dont want it to mess with the look of the layout on mobile, it will be removed and add hamburger on mobile view

Comment (larrycoal) [reply src/index.css]: The logo isnt the official one, downloaded the logo online

File: src/pages/resources/Resources.js (hunk 9 above)
Comment (larrycoal) [reply src/pages/resources/Resources.js]: can you clarify on what you mean by inlining

Comment (ANDREYDEN) [reply src/pages/resources/Resources.js]: Inlining is the opposite of extracting. In your original function you extracted everything related to rendering resources (including the `.map()`) into a separate function. I'm just suggesting you keep the map inlined (inside the component code) and extract the function that renders the individual resource.

So here's the change that I want you to do:
```suggestion
          // your function at the top of the file
          function renderResource(resource, idx) => {
            return (
              <li key={idx}>
                <div className="image">
                  <img src={resource.logo} alt={resource.name} />
                </div>
                <div className="header">{resource.name}</div>
                <div>{resource.description}</div>
                <div className="link">
                  <a href={resource.link} target="_blank" rel="noreferrer">
                    {resource.name}
                  </a>
                </div>
              </li>
            )
          }
          
          // the usage in the component
          {resources.map(renderResource)}
```

File: src/index.css (hunk 7 above)
Comment (ANDREYDEN) [reply src/index.css]: I'm sorry about being nitpicky, but can we both keep the mobile layout and not show the scroll bar on desktop?

Comment (ANDREYDEN) [reply src/index.css]: please make sure to use the [official](https://www.youtube.com/howyoutubeworks/resources/brand-resources/#logos-icons-and-colors) logos

Comment (larrycoal) [reply src/index.css]: sure i can remove the overflow rule that made the header have a scroll bar,but to make the page properly responsive i would need to work on the header component, should i do that, its outside of the scope of my work so i need the go ahead to work on the header

Comment (ANDREYDEN) [reply src/index.css]: Yes, the menu component is out of scope of this issue, so it would've been better if you could open another issue just for this. How about you hide the logo on mobile, would that solve the issue?
```
@media only screen and (max-width: 500px) {
  .menu-header {
    display: none;
  }
}
```

Comment (larrycoal) [reply src/index.css]: i can open a new issue for the header component and have a proper display for mobile screen, i think hiding the logo is a temporary fix

**Review by ANDREYDEN — CHANGES_REQUESTED** [review]
A few corrections:
- The icons should be the official brand icons downloaded from the brand portals:
  - [YouTube](https://www.youtube.com/howyoutubeworks/resources/brand-resources/#logos-icons-and-colors) - make sure to use the **icons**
  - [LinkedIn](https://brand.linkedin.com/downloads) - make sure to use the **icons** (without the word "LinkedIn")
  - [GitHub](https://github.com/logos)
  - [Instagram](https://www.facebook.com/brand/resources/instagram/instagram-brand/)
  - [Twitter](https://about.twitter.com/en/who-we-are/brand-toolkit)
  - [Facebook](https://www.facebook.com/brand/resources/facebookapp/logo) - not Meta, because Meta is the parent company that owns both Facebook and Instagram
- All icons should maintain their ratio (width x height) and not be skewed on different screen sizes
- Please, use the icons that most resemble a square (for YouTube and LinkedIn they should not include the names of the services - just the logos)
- do not include the `.zip` files in source control - those are unneccesary and increase build and load times. Only include the `png`s of the logos

File: src/index.css (hunk 7 above)
Comment (ANDREYDEN) [reply src/index.css]: yes, I agree. If you want to revert your changes to the `Menu` component in this PR and make them in a separate one (tied to a new issue) - that would be ideal

Comment (larrycoal) [reply src/index.css]: okay

File: src/pages/resources/Resources.css [hunk 10]
```diff
@@ -1,3 +1,59 @@
...
+  padding: 2rem;
+}
+.resources_list li {
+  width: 30%;
+  padding: 10px;
+  border-radius: 20px;
+  box-shadow: 1px 0px 9px -2px #5b6372;
+}
+.resources_list li img {
+  height: 120px;
```
Comment (ANDREYDEN) [inline src/pages/resources/Resources.css]: Let's only constrain the width and leave height dynamic as not all logo's have the 14x12 ratio
```suggestion
```

**Review by ANDREYDEN — CHANGES_REQUESTED** [review]
Much better! Just one last tweak

File: src/pages/resources/Resources.css (hunk 10 above)
Comment (larrycoal) [reply src/pages/resources/Resources.css]: done

**Review by ANDREYDEN — APPROVED** [review]
Good to go!
